### PR TITLE
Added torrent-file-editor manifest

### PR DESCRIPTION
**Pull request by FacundoAcevedo**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **[Link:](https://github.com/torrent-file-editor/torrent-file-editor/issues/133)**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.Torrent-file-editor.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <li>Create .torrent file from scratch</li>
+    <li>Edit .torrent file in user-friendly way</li>
+    <li>Edit .torrent file in JSON format</li>
+    <li>Add, remove and interchange files in .torrent file</li>
+    <li>Support for codings</li>
+    <li xml:lang="ru">Создание и редактирование .torrent файлов с нуля</li>
+    <li xml:lang="ru">Редактирование .torrent файлов в формате древовидной структуры</li>
+    <li xml:lang="ru">Редактирование .torrent файлов в JSON-формате</li>
+    <li xml:lang="ru">Добавление и удаление файлов, а также изменение их порядка в .torrent файле</li>
+    <li xml:lang="ru">Поддержка кодировок</li>
```
Comment (JakobDev) [inline io.github.Torrent-file-editor.metainfo.xml:33]:
```suggestion
    <li>Create .torrent file from scratch</li>
    <li xml:lang="ru">Создание и редактирование .torrent файлов с нуля</li>
    <li>Edit .torrent file in user-friendly way</li>
    <li xml:lang="ru">Редактирование .torrent файлов в формате древовидной структуры</li>
    <li>Edit .torrent file in JSON format</li>
    <li xml:lang="ru">Редактирование .torrent файлов в JSON-формате</li>
    <li>Add, remove and interchange files in .torrent file</li>
    <li xml:lang="ru">Добавление и удаление файлов, а также изменение их порядка в .torrent файле</li>
    <li>Support for codings</li>
    <li xml:lang="ru">Поддержка кодировок</li>
```

File: io.github.Torrent-file-editor.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/torrent-file-editor/torrent-file-editor/releases/latest
+          version-query: '.tag_name | sub("^v"; "")'
+          timestamp-query: .published_at
+          url-query: '.assets[] | select(.name=="torrent-file-editor-" + $version + ".tar.gz") | .browser_download_url'
+      - type: file
+        path: io.github.Torrent-file-editor.metainfo.xml
+      - type: file
+        path: io.github.Torrent-file-editor.desktop
```
Comment (JakobDev) [inline io.github.Torrent-file-editor.yaml:22]: This files doesn't seems to be installed. They also belongs upstream

File: io.github.Torrent-file-editor.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>torrent-file-editor.desktop</id>
```
Comment (JakobDev) [inline io.github.Torrent-file-editor.metainfo.xml:3]:
```suggestion
  <id>torrent-file-editor</id>
```

**Review by JakobDev — COMMENTED** [review]
You also need to add
```xml
<launchable type="desktop-id">torrent-file-editor.desktop</launchable>
```
to your metainfo

**Comment by hfiguiere** [comment]
> You also need to add
> 
> ```
> <launchable type="desktop-id">torrent-file-editor.desktop</launchable>
> ```
> 
> to your metainfo

no. it's not necessary. if it is not here, appstream compose is fine. Also if adding it, why using the wrong value like you suggest?

File: io.github.Torrent-file-editor.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+app-id: io.github.torrent-file-editor
```
Comment (hfiguiere) [inline io.github.Torrent-file-editor.yaml:1]: this app-id doesn't meet the requirements.

File: io.github.Torrent-file-editor.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+runtime: org.kde.Platform
+runtime-version: "5.15-21.08"
+sdk: org.kde.Sdk
+
+command: torrent-file-editor
+modules:
+  - name: torrent-file-editor
+    sources:
```
Comment (hfiguiere) [inline io.github.Torrent-file-editor.yaml:9]: source is usually last in the module

File: io.github.Torrent-file-editor.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    buildsystem: "cmake"
+    config-opts:
+      #- -DCMAKE_BUILD_TYPE=Release
+      - -DCMAKE_INSTALL_LIBDIR=/app/lib
+      - -DLIB_INSTALL_DIR=/app/lib
+      - -DQT5_BUILD=ON
+      - -DENABLE_PCH=OFF
+      - -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON
+
+finish-args:
```
Comment (hfiguiere) [inline io.github.Torrent-file-editor.yaml:32]: finish-args is usually above the modules.

File: io.github.Torrent-file-editor.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+          timestamp-query: .published_at
+          url-query: '.assets[] | select(.name=="torrent-file-editor-" + $version + ".tar.gz") | .browser_download_url'
+      - type: file
+        path: io.github.Torrent-file-editor.metainfo.xml
+      - type: file
+        path: io.github.Torrent-file-editor.desktop
+    buildsystem: "cmake"
+    config-opts:
+      #- -DCMAKE_BUILD_TYPE=Release
+      - -DCMAKE_INSTALL_LIBDIR=/app/lib
```
Comment (hfiguiere) [inline io.github.Torrent-file-editor.yaml:26]:
```suggestion
```

This is not necessary

File: io.github.Torrent-file-editor.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+          type: json
+          url: https://api.github.com/repos/torrent-file-editor/torrent-file-editor/releases/latest
+          version-query: '.tag_name | sub("^v"; "")'
+          timestamp-query: .published_at
+          url-query: '.assets[] | select(.name=="torrent-file-editor-" + $version + ".tar.gz") | .browser_download_url'
+      - type: file
+        path: io.github.Torrent-file-editor.metainfo.xml
+      - type: file
+        path: io.github.Torrent-file-editor.desktop
+    buildsystem: "cmake"
```
Comment (hfiguiere) [inline io.github.Torrent-file-editor.yaml:23]:
```suggestion
    buildsystem: "cmake-ninja"
```

File: io.github.Torrent-file-editor.desktop [hunk 9]
```diff
@@ -0,0 +1,14 @@
+[Desktop Entry]
+Categories=Network;P2P;Qt;
+Exec=torrent-file-editor %f
+GenericName=Torrent File Editor
+Comment=Edit and create .torrent files
+Icon=torrent-file-editor
```
Comment (hfiguiere) [inline io.github.Torrent-file-editor.desktop:6]: this icon name is wrong. Given that you don't do a `rename-icon`

File: io.github.Torrent-file-editor.metainfo.xml (hunk 3 above)
Comment (hfiguiere) [reply io.github.Torrent-file-editor.metainfo.xml:3]: No. This should be the actuall app-id. There is no point changing a wrong to another wrong.

File: io.github.Torrent-file-editor.yaml (hunk 4 above)
Comment (hfiguiere) [reply io.github.Torrent-file-editor.yaml:1]: see https://docs.flatpak.org/en/latest/conventions.html#application-ids

File: io.github.Torrent-file-editor.yaml (hunk 2 above)
Comment (FacundoAcevedo) [reply io.github.Torrent-file-editor.yaml]: I removed the lines and used the ones from upstream
https://github.com/flathub/flathub/pull/3457/commits/1f5d8b6cf2fd43b62c1ced32518d69c317311869

File: io.github.Torrent-file-editor.metainfo.xml (hunk 3 above)
Comment (FacundoAcevedo) [reply io.github.Torrent-file-editor.metainfo.xml]: fixed
https://github.com/flathub/flathub/pull/3457/files#diff-1ec4941b291ac3dc01816e41751d303a4b244014c2a9aef4b4d45df1cdc37290R9

File: io.github.Torrent-file-editor.yaml (hunk 4 above)
Comment (FacundoAcevedo) [reply io.github.Torrent-file-editor.yaml]: Fixed I also renamed the yaml file

File: io.github.Torrent-file-editor.yaml (hunk 5 above)
Comment (FacundoAcevedo) [reply io.github.Torrent-file-editor.yaml]: fixed here: https://github.com/flathub/flathub/pull/3457/commits/69ef24cf14a8e456ca2b1fa93cb076f62e12d3bf

File: io.github.Torrent-file-editor.yaml (hunk 6 above)
Comment (FacundoAcevedo) [reply io.github.Torrent-file-editor.yaml]: fixed here: https://github.com/flathub/flathub/pull/3457/commits/69ef24cf14a8e456ca2b1fa93cb076f62e12d3bf

File: io.github.torrent_file_editor.Torrent-file-editor.yaml [hunk 10]
```diff
@@ -0,0 +1,41 @@
+app-id: io.github.torrent_file_editor.torrent-file-editor
```
Comment (hfiguiere) [inline io.github.torrent_file_editor.Torrent-file-editor.yaml:1]: this is case sensitive and doesn't match the filename or the patch.

File: io.github.torrent_file_editor.Torrent-file-editor.yaml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  - --share=ipc
+  - --socket=fallback-x11
+  #Wayland
+  - --socket=wayland
+  - --device=dri
+modules:
+  - name: torrent-file-editor
+    buildsystem: "cmake-ninja"
+    config-opts:
+      #- -DCMAKE_BUILD_TYPE=Release
```
Comment (hfiguiere) [inline io.github.torrent_file_editor.Torrent-file-editor.yaml:23]:
```suggestion
```

File: io.github.Torrent-file-editor.yaml (hunk 7 above)
Comment (FacundoAcevedo) [reply io.github.Torrent-file-editor.yaml]: fixed here: https://github.com/flathub/flathub/pull/3457/commits/72b76e51b2b55ae47f7296e3dcbc87a555ab704b

File: io.github.Torrent-file-editor.yaml (hunk 8 above)
Comment (FacundoAcevedo) [reply io.github.Torrent-file-editor.yaml]: fixed

File: io.github.torrent_file_editor.Torrent-file-editor.yaml [hunk 12]
```diff
@@ -0,0 +1,41 @@
+app-id: io.github.torrent_file_editor.torrent-file-editor
+runtime: org.kde.Platform
+runtime-version: "5.15-21.08"
+sdk: org.kde.Sdk
+
+rename-desktop-file: torrent-file-editor.desktop
+rename-appdata-file: torrent-file-editor.appdata.xml
+#rename-icon: torrent-file-editor.png
```
Comment (hfiguiere) [inline io.github.torrent_file_editor.Torrent-file-editor.yaml:8]:
```suggestion
rename-icon: torrent-file-editor
```

you likely need this.

Comment (FacundoAcevedo) [reply io.github.torrent_file_editor.Torrent-file-editor.yaml:8]: thank you very much I was having a bad time with that line

File: io.github.torrent_file_editor.Torrent-file-editor.yaml (hunk 10 above)
Comment (FacundoAcevedo) [reply io.github.torrent_file_editor.Torrent-file-editor.yaml]: fixed

File: io.github.torrent_file_editor.Torrent-file-editor.yaml (hunk 11 above)
Comment (FacundoAcevedo) [reply io.github.torrent_file_editor.Torrent-file-editor.yaml]: fixed

**Comment by hfiguiere** [comment]
bot, build io.github.torrent_file_editor.Torrent-file-editor

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/io.github.torrent_file_editor.Torrent-file-editor.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://a.fsdn.com/con/app/proj/torrent-file-editor/screenshots/Selection_113.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://a.fsdn.com/con/app/proj/torrent-file-editor/screenshots/Selection_114.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://a.fsdn.com/con/app/proj/torrent-file-editor/screenshots/Selection_116.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://a.fsdn.com/con/app/proj/torrent-file-editor/screenshots/Selection_115.png]
• tag-missing           : <release> required
Validation of files failed
```

**Comment by hfiguiere** [comment]
bot, build io.github.torrent_file_editor.Torrent-file-editor

**Comment by FacundoAcevedo** [comment]
Thank you very much for the review I learned a lot :D

**Pull request closed without merging**